org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/Utilities.java

```diff
@@ -1308,6 +1308,19 @@ public static boolean startsWithInList(String s, Collection<String> list) {
     return false;
   }
 
+  public static String describeSize(int length) {
+    if (length > 1024 * 1024 * 1024) {
+      return ""+length / (1024 * 1024 * 1024)+"Gb";
+    }
+    if (length > 1024 * 1024) {
+      return ""+length / (1024 * 1024)+"Mb";
+    }
+    if (length > 1024) {
+      return ""+length / (1024)+"kb";
+    }
+    return ""+length +"b";
+  }
+
```

Comment:
Can we change the code for this method to this:

```
    public static final int ONE_MB = 1024;
    public static final String GB = "Gb";
    public static final String MB = "Mb";
    public static final String KB = "Kb";
    public static final String BT = "b";

    public static String describeSize(int length) {
        if (length < 0) throw new IllegalArgumentException("File length of < 0  passed in...");

        if (length > Math.pow(ONE_MB, 3)) {
            return length / ((long) Math.pow(ONE_MB, 3)) + GB;
        }
        if (length > Math.pow(ONE_MB, 2)) {
            return length / ((long) Math.pow(ONE_MB, 2)) + MB;
        }
        if (length > ONE_MB) {
            return length / (ONE_MB) + KB;
        }
        return length + BT;
    }
```

...and this kind of utility method lends itself very well to testing, so we can add a test method like so:

```
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.Random;

public class testTest {

    public static final int BOUND = 500;
    public static final Random RAND = new Random();

    public static final int GB_MEASURE_JUST_OVER = (int) Math.pow(Utilities.ONE_MB, 3) + RAND.nextInt(BOUND);
    public static final int GB_MEASURE_EXACT = (int) Math.pow(Utilities.ONE_MB, 3);
    public static final int GB_MEASURE_JUST_UNDER = (int) Math.pow(Utilities.ONE_MB, 3) - RAND.nextInt(BOUND);

    public static final int MB_MEASURE_JUST_OVER = (int) Math.pow(Utilities.ONE_MB, 2) + RAND.nextInt(BOUND);
    public static final int MB_MEASURE_EXACT = (int) Math.pow(Utilities.ONE_MB, 2);
    public static final int MB_MEASURE_JUST_UNDER = (int) Math.pow(Utilities.ONE_MB, 2) - RAND.nextInt(BOUND);

    public static final int KB_MEASURE_JUST_OVER = Utilities.ONE_MB + RAND.nextInt(BOUND);
    public static final int KB_MEASURE_EXACT = Utilities.ONE_MB;
    public static final int KB_MEASURE_JUST_UNDER = Utilities.ONE_MB - RAND.nextInt(BOUND);

    public static final int BT_MEASURE = Utilities.ONE_MB + RAND.nextInt(BOUND);
    public static final int EMPTY = 0;

    public static final int BIG_NEG = Utilities.ONE_MB * -1;

    @Test
    @DisplayName("Test size bounds on file size utility.")
    void describeSizeTest() {
        Assertions.assertAll("GB Measure Limits",
                () -> assertTrue(Utilities.describeSize(GB_MEASURE_JUST_OVER).contains(Utilities.GB)),
                () -> assertTrue(Utilities.describeSize(GB_MEASURE_EXACT).contains(Utilities.MB)),
                () -> assertTrue(Utilities.describeSize(GB_MEASURE_JUST_UNDER).contains(Utilities.MB))
        );
        Assertions.assertAll("MB Measure Limits",
                () -> assertTrue(Utilities.describeSize(MB_MEASURE_JUST_OVER).contains(Utilities.MB)),
                () -> assertTrue(Utilities.describeSize(MB_MEASURE_EXACT).contains(Utilities.KB)),
                () -> assertTrue(Utilities.describeSize(MB_MEASURE_JUST_UNDER).contains(Utilities.KB))
        );
        Assertions.assertAll("KB Measure Limits",
                () -> assertTrue(Utilities.describeSize(KB_MEASURE_JUST_OVER).contains(Utilities.KB)),
                () -> assertTrue(Utilities.describeSize(KB_MEASURE_EXACT).contains(Utilities.BT)),
                () -> assertTrue(Utilities.describeSize(KB_MEASURE_JUST_UNDER).contains(Utilities.BT))
        );
        Assertions.assertAll("BT Measure Limits",
                () -> assertTrue(Utilities.describeSize(BT_MEASURE).contains(Utilities.BT)),
                () -> assertTrue(Utilities.describeSize(EMPTY).contains(Utilities.BT))
        );
        Assertions.assertThrows(IllegalArgumentException.class, () -> Utilities.describeSize(BIG_NEG));
    }
}
```

This is running under the assumption that:
a) We return the measure in the next lowest unit if the measurement is exactly on the boundary (ie, 1024b, as opposed to 1Kb)
b) File sizes of 0 are valid
c) Negative file sizes result in an IllegalArgumentException being thrown

Make any changes to those you want, but we need the test here.